raft.go

```diff
@@ -1790,15 +1800,20 @@ func (r *Raft) installSnapshot(rpc RPC, req *InstallSnapshotRequest) {
 	r.setLatestConfiguration(reqConfiguration, reqConfigurationIndex)
 	r.setCommittedConfiguration(reqConfiguration, reqConfigurationIndex)
 
-	// Compact logs, continue even if this fails
-	if err := r.compactLogs(req.LastLogIndex); err != nil {
-		r.logger.Error("failed to compact logs", "error", err)
+	// Fully truncate logs for stores with monotonically increasing log indexes.
+	// Otherwise, compact the logs. Log any errors and continue.
+	err = r.resetMonotonicLogs()
```

Comment:
Using an `error` as a signal for a non-error case isn't ideal. I wonder if it's cleaner just to do the type assertion here?

If this saves a significant amount of duplication a more idiomatic way would be to change `resetMonotonicLogs` to return a bool that indicates whether it did anything or not rather than an error. If there are legit reasons that this could error too then return both, but in that case we should also handle the error up on line 1133 (even if we choose to swallow it or just log it etc.)